frontend/src/metabase/query_builder/actions/core/updateQuestion.ts

```diff
@@ -129,17 +130,23 @@ export const updateQuestion = (
 
       // When the dataset query changes, we should change the question type,
       // to start building a new ad-hoc question based on a dataset
-      if (newQuestion.isDataset()) {
+      if (newQuestion.type() === "model" || newQuestion.type() === "metric") {
         newQuestion = newQuestion.setType("question");
         dispatch(onCloseQuestionInfo());
       }
     }
 
-    // This scenario happens because the DatasetQueryEditor converts the dataset/model question into a normal question
+    // This scenario happens because the DatasetQueryEditor converts the model/metric question into a normal question
```

Comment:
```suggestion
    // This scenario happens because the DatasetQueryEditor converts a model or a metric into a normal question
```

The previous change could be easily misinterpreted.